### PR TITLE
features/edgevit3

### PR DESCRIPTION
**Pull request by Jiabei-prog**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

# Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.

# Modification
Please briefly describe what modification is made in this PR.

# Backward-Compatibility-Breaking (Optional)
Does the modification introduce changes that break the backward-compatibility? If so, please describe how it breaks the compatibility in this PR.

# Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

# Pull-Request Todo List
1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+log_config = dict(
+    interval=10,
+    hooks=[dict(type='TextLoggerHook'),
+           dict(type='TensorboardLoggerHook')])
+
+# model settings
+model = dict(
+    type='Classification',
+    # train_preprocess=['mixUp'],
```
Comment (Cathy0908) [inline configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py:11]: 删除注释掉的无用代码

File: configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        in_channels=288,
+        loss_config=dict(
+            type='CrossEntropyLossWithLabelSmooth', label_smooth=0),
+    ))
+
+data_train_list = 'data/imagenet_raw/meta/train_labeled.txt'
+data_train_root = 'data/imagenet_raw/train/'
+data_test_list = 'data/imagenet_raw/meta/val_labeled.txt'
+data_test_root = 'data/imagenet_raw/validation/'
+data_all_list = 'data/imagenet_raw/meta/all_labeled.txt'
```
Comment (Cathy0908) [inline configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py:43]: data_all_list data_root没用到，删除

File: configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        type='MAEFtAugment',
+        input_size=224,
+        mean=(0.485, 0.456, 0.406),
+        std=(0.229, 0.224, 0.225),
+        is_train=False,
+    ),
+]
+
+data = dict(
+    imgs_per_gpu=1024,  # total 256
```
Comment (Cathy0908) [inline configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py:73]: total 256？

File: easycv/models/backbones/edgevit.py [hunk 4]
```diff
@@ -0,0 +1,469 @@
+import math
```
Comment (Cathy0908) [inline easycv/models/backbones/edgevit.py:1]: add license

File: easycv/models/backbones/edgevit.py [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+from collections import OrderedDict
+from functools import partial
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils.checkpoint as checkpoint
+from mmcv.cnn import build_norm_layer, constant_init, kaiming_init
```
Comment (Cathy0908) [inline easycv/models/backbones/edgevit.py:9]: from easycv.models.utils.norm import build_norm_layer  对比下easycv的接口

File: easycv/models/backbones/edgevit.py [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+from timm.models.vision_transformer import _cfg
+from torch.nn.modules.batchnorm import _BatchNorm
+
+from easycv.utils.checkpoint import load_checkpoint
+from easycv.utils.logger import get_root_logger
+from ..registry import BACKBONES
+from ..utils import build_conv_layer
+
+
+class Mlp(nn.Module):
```
Comment (Cathy0908) [inline easycv/models/backbones/edgevit.py:21]: 和easycv/models/utils/transformer.py中的Mlp接口重复

File: easycv/models/backbones/edgevit.py [hunk 7]
```diff
@@ -0,0 +1,469 @@
...
+    def forward(self, x):
+        x = self.fc1(x)
+        x = self.act(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+        x = self.drop(x)
+        return x
+
+
+class CMlp(nn.Module):
```
Comment (Cathy0908) [inline easycv/models/backbones/edgevit.py:46]: 和`easycv/models/backbones/conv_mae_vit.py`中的`ConvMlp`接口重复，可以将该接口放到`easycv/models/utils/transformer.py`中，作为通用工具

**Review by Cathy0908 — COMMENTED** [review]
add unittest

File: configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py (hunk 1 above)
Comment (Jiabei-prog) [reply configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py]: 已删除

File: configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py (hunk 2 above)
Comment (Jiabei-prog) [reply configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py]: 已删除

File: easycv/models/backbones/edgevit.py (hunk 4 above)
Comment (Jiabei-prog) [reply easycv/models/backbones/edgevit.py]: 已加license

File: easycv/models/backbones/edgevit.py (hunk 5 above)
Comment (Jiabei-prog) [reply easycv/models/backbones/edgevit.py]: build_norm_layer没有用到，已删除

File: easycv/models/backbones/edgevit.py (hunk 7 above)
Comment (Jiabei-prog) [reply easycv/models/backbones/edgevit.py]: 在tranformer里加入了ConvMlp

File: docs/source/model_zoo_cls.md [hunk 8]
```diff
@@ -82,5 +82,6 @@
...
 | efficientformer_l3 | [efficientformer_l3](https://github.com/alibaba/EasyCV/tree/master/configs/classification/imagenet/efficientformer/efficientformer_l3.py) | 82.272   | 96.028   | 2436    | 13.07    | [model](https://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/classification/efficientformer/efficientformer_l3_300d.pth) |
 | efficientformer_l7 | [efficientformer_l7](https://github.com/alibaba/EasyCV/tree/master/configs/classification/imagenet/efficientformer/efficientformer_l7.py) | 83.076   | 96.44   | 1622    | 18.96    | [model](https://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/classification/efficientformer/efficientformer_l7_300d.pth) |
+| EdgeVit_xxs_b512_224 | [EdgeVit_xxs_b512_224](https://github.com/alibaba/EasyCV/tree/master/configs/classification/imagenet/edgevit/EdgeVit_b512x8_300e_jpg.py) | 75.18   | 92.19   | 13876    | 8.632    | [model](http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/classification/edgevit/edgexxs/ClsEvaluator_neck_top1_best.pth) |
```
Comment (wenmengzhou) [inline docs/source/model_zoo_cls.md:85]: why edgevit xxs is slower than  vit base？

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**